security/pkg/nodeagent/cache/helper.go

```diff
@@ -78,3 +74,31 @@ func cacheLogPrefix(conID, resourceName string) string {
 	lPrefix := fmt.Sprintf("CONNECTION ID: %s, RESOURCE NAME: %s, EVENT:", conID, resourceName)
 	return lPrefix
 }
+
+// isTrustworthyJwt checks if a jwt is a trustworthy jwt type.
+func isTrustworthyJwt(jwt string) (bool, error) {
+	type trustWorthyJwtPayload struct {
+		Aud []string `json:"aud"`
+		Exp int      `json:"exp"`
+	}
+
+	jwtSplit := strings.Split(jwt, ".")
+	if len(jwtSplit) != 3 {
+		return false, fmt.Errorf("jwt may be invalid: %s", jwt)
+	}
+	payload := jwtSplit[1]
+
+	payloadBytes, err := base64.RawStdEncoding.DecodeString(payload)
+	if err != nil {
+		return false, fmt.Errorf("failed to decode jwt: %v", err.Error())
```

Comment:
log the jwt here as well?

Reply:
We actually log it here https://github.com/istio/istio/blob/d155cdc40a402347e2aba9295fd7205d1ab4287e/security/pkg/nodeagent/cache/secretcache.go#L595, but I agree, we should log it in the callee.